sceptre/cli/dump.py

```diff
@@ -48,8 +52,70 @@ def dump_config(ctx, path):
 
     output_format = "json" if context.output_format == "json" else "yaml"
 
+    def process_config(config):
+        stack_name = list(config.keys())[0]
+        file_path = Path(f".dump/{stack_name}/config.yaml")
+        write(config[stack_name], output_format, no_colour=True, file_path=file_path)
+
     if len(output) == 1:
-        write(output[0][stack.external_name], output_format)
+        process_config(output[0])
     else:
         for config in output:
-            write(config, output_format)
+            process_config(config)
```

Comment:
I don't see a reason any more to have this `if/else` here. I think you could just have the for loop.

Reply:
`process_config(output[0])` is the same as doing this: 

```python
for config in output[:1]:
    process_config(output)
```

And if output is a list of only one item, it's the same as doing this, because it'll only loop once:

```python
for config in output:
    process_config(output)
```

Because remember, looping through a list like this is essentially the same as doing:

```python
for i in range(len(output)):
    process_config(output[i])
```

But if that's the flow you want for >1 as well, you can just use that for output, no matter if there are 0 items (it will loop zero times), 1 items (it will loop once), or 100 items. 

Anyway, it's all the same. You don't need the if/else.

Reply:
@jfalkenstein ah! I see your point. They used to do different things and now they don't. Updated.